### PR TITLE
Use @embroider/test-setup and get embroider passing

### PR DESCRIPTION
**Pull request by thoov**
Fixes https://github.com/ember-cli/ember-exam/issues/634

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -82,19 +86,23 @@
...
     "rsvp": "^4.8.5",
     "sinon": "^9.1.0",
     "testdouble": "^3.16.1"
   },
   "peerDependencies": {
+    "@ember/test-helpers": "*",
     "ember-mocha": "*",
-    "ember-qunit": "*"
+    "ember-qunit": "*",
+    "qunit": "*"
```
Comment (rwjblue) [inline package.json:105]: Can you add this to peerDependenciesMeta to make it optional?

**Review by rwjblue — COMMENTED** [review]
Can you add another scenario that uses ember-qunit@4?

File: package.json (hunk 1 above)
Comment (thoov) [reply package.json:104]: done

File: package.json [hunk 2]
```diff
@@ -70,7 +74,7 @@
     "ember-disable-prototype-extensions": "^1.1.2",
     "ember-export-application-global": "^2.0.1",
     "ember-load-initializers": "^2.1.1",
-    "ember-qunit": "^4.6.0",
+    "ember-qunit": "^5.1.2",
```
[the remark this answers is not in the document]
Comment (thoov) [reply package.json:77]: done

File: package.json [hunk 3]
```diff
@@ -86,15 +90,18 @@
...
     "nyc": "^15.1.0",
+    "qunit": "^2.14.0",
     "release-it": "^14.0.2",
     "release-it-lerna-changelog": "^2.4.0",
     "rsvp": "^4.8.5",
     "sinon": "^9.1.0",
     "testdouble": "^3.16.1"
   },
   "peerDependencies": {
+    "@ember/test-helpers": "*",
```
[the remark this answers is not in the document]
Comment (thoov) [reply package.json]: done

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -43,16 +43,18 @@ jobs:
...
           - embroider-optimized
           - embroider-optimized-with-mocha
+          - ember-lts-3.24
           - ember-lts-3.20
           - ember-lts-3.16
           - ember-lts-3.12
           - ember-release
           - ember-beta
           - ember-canary
+          - ember-qunit-4
```
Comment (rwjblue) [inline .github/workflows/ci.yml:57]: 👍 thanks for adding this

**Review by rwjblue — APPROVED** [review]

**Pull request merged**